libnavigation-core/src/test/java/com/mapbox/navigation/core/routeoptions/NavRouteOptionsBuilderTest.kt

```diff
@@ -0,0 +1,273 @@
+                name = "test name",
+                bearing = 45.0,
+                zLevel = 6
+            )
+            .build()
+        assertEquals(
+            "test name",
+            options.waypointNamesList()?.get(1)
+        )
+        assertEquals(
+            45.0,
+            options.bearingsList()!![1].angle(),
+            0.01
+        )
+        assertEquals(
+            6,
+            options.layersList()!![1]
+        )
+    }
+
+    @Test
+    fun `add silent waypoint`() = runBlocking<Unit> {
+        val options = createBuilder()
+            .applyRequiredOptionsForTesting()
+            .addSilentWaypoint(
+                coordinate = Point.fromLngLat(1.0, 1.0),
+                bearing = 45.0,
+                zLevel = 6
+            )
+            .build()
+
+        assertEquals(3, options.coordinatesList().size)
+        assertEquals(
+            Point.fromLngLat(1.0, 1.0),
+            options.coordinatesList()[1]
+        )
+        assertEquals(
+            45.0,
+            options.bearingsList()!![1].angle(),
+            0.01
+        )
+        assertEquals(
+            6,
+            options.layersList()!![1]
+        )
+        assertEquals(
+            listOf(0, 2),
+            options.waypointIndicesList()
+        )
+    }
+
+    @Test
+    fun `mix silent and named waypoints`() = runBlocking<Unit> {
+        val options = createBuilder()
+            .applyRequiredOptionsForTesting()
+            .addSilentWaypoint(
+                coordinate = Point.fromLngLat(1.0, 1.0)
+            )
+            .addIntermediateWaypoint(
+                coordinate = Point.fromLngLat(2.0, 2.0),
+                name = "test"
+            )
+            .build()
+
+        assertEquals(
+            listOf(null, "test", null),
+            options.waypointNamesList()
+        )
+    }
+
+    @Test
+    fun `driving profile with excluded toll and unpaved`() = runBlocking<Unit> {
+        val options = createBuilder()
+            .fromCurrentPosition()
+            .applyTestDestination()
+            .profileDriving {
+                exclude {
+                    toll()
+                    unpaved()
+                }
+            }
+            .build()
+
+        assertEquals(DirectionsCriteria.PROFILE_DRIVING, options.profile())
+        assertEquals(
+            listOf(
+                DirectionsCriteria.EXCLUDE_TOLL,
+                DirectionsCriteria.EXCLUDE_UNPAVED
+            ),
+            options.excludeList()
+        )
+    }
+
+    @Test
+    fun `driving-traffic profile with excluded motorway, ferry, and cash only tolls`() = runBlocking<Unit> {
+        val options = createBuilder()
+            .fromCurrentPosition()
+            .applyTestDestination()
+            .profileDrivingTraffic {
+                exclude {
+                    motorway()
+                    ferry()
+                    cashOnlyTolls()
+                }
+            }
```

Comment:
Do we support usage of the SDK from java? Not sure if that will work great for java 🤔

Reply:
Not so bad. See java unit tests